mrmustard/utils/training.py

```diff
@@ -85,10 +90,20 @@ def should_stop(self, max_steps: int) -> bool:
                 sum(abs(self.opt_history[-i - 1] - self.opt_history[-i]) for i in range(1, 20))
                 < 1e-6
             ):
-                print("Loss looks stable, stopping here.")
+                self.log.info("Loss looks stable, stopping here.")
                 return True
         return False
 
+    class OptimizerInterruptedError(Exception):
+        """A helper class to quietly stop execution without printing a traceback."""
+
+        def __init__(self, message: str, logger: Logger) -> None:
+            super().__init__()
+            logger.info(message)
+
+        def _render_traceback_(self):
+            pass
```

Comment:
Is this actually needed when raising exception from `None`? 🤔

Reply:
Raising from `None` still raises a traceback which is quite uninformative since the `KeyboardInterrup` exception is swallowed by raising `None`. The empty [`_render_traceback_`](https://ipython.readthedocs.io/en/stable/config/integrating.html#custom-exception-tracebacks) avoids this traceback to be printed out on ipy-interpreters, hence giving a less clumped message when stopping execution.